### PR TITLE
Let it snow2

### PR DESCRIPTION
**Pull request by Gymnasiast**
This is a rebased version of https://github.com/OpenRCT2/OpenRCT2/pull/10754.

I have taken out the weather transitions, as they led to it snowing in _July_. That bit needs a rethink. In the meantime, though, people can now use the "Force Weather" feature to set their park to use snow.

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Sounds like a good compromise. Perhaps we could introduce colder climates in the new save format.

**Comment by Gymnasiast** [comment]
Either that or making sure snow can only happen in March, April, September, October (cold climate) or March, October (cool and wet).

File: distribution/changelog.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@
 - Feature: [#12712] Add TCP / socket plugin APIs.
 - Feature: [#12840] Add Park.entranceFee to the plugin API.
 - Feature: [#12886] Make all scenery placement and remove actions available to the plugin API.
+- Feature: [#12922] Add snow, heavy snow and blizzard to weather types.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:8]: add #2350 and #10754

Comment (AaronVanGeffen) [reply distribution/changelog.txt:8]: I think just mentioning #2350 should do.

**Comment by brettpenzer123** [comment]
Glad to see it's been added sadly I don't have enough time to work on my own projects and open-source ones as well. I did try to reimplement it as much as possible from my first initial implementation way back. I'm also not involved in the project as much as I'd like to be to which come to a lot of discussion about certain types of weather. Hope someone makes good use of it and good luck to those that carry this on. 👍 

My city is going back into a local lockdown so this should be fun 🗡️

File: src/openrct2-ui/drawing/engines/opengl/OpenGLDrawingEngine.cpp [hunk 2]
```diff
@@ -121,20 +122,21 @@ class OpenGLDrawingContext final : public IDrawingContext
...
-    explicit OpenGLRainDrawer(OpenGLDrawingContext* drawingContext)
+    explicit OpenGLWeatherDrawer(OpenGLDrawingContext* drawingContext)
         : _drawingContext(drawingContext)
     {
     }
 
-    virtual void Draw(int32_t x, int32_t y, int32_t width, int32_t height, int32_t xStart, int32_t yStart) override
+    virtual void Draw(
+        int32_t x, int32_t y, int32_t width, int32_t height, int32_t xStart, int32_t yStart,
+        const uint8_t* weatherpattern) override
```
Comment (Gymnasiast) [inline src/openrct2-ui/drawing/engines/opengl/OpenGLDrawingEngine.cpp:137]: check this

File: src/openrct2/drawing/Weather.cpp [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*****************************************************************************
- * Copyright (c) 2014-2020 OpenRCT2 developers
+ * Copyright (c) 2014-2019 OpenRCT2 developers
```
Comment (Gymnasiast) [inline src/openrct2/drawing/Weather.cpp:2]: revert

File: src/openrct2/world/Climate.cpp [hunk 4]
```diff
@@ -411,24 +438,30 @@ const FILTER_PALETTE_ID ClimateWeatherGloomColours[4] = {
...
+    { 2, WeatherEffectType::Storm, 2, WeatherLevel::Heavy, SPR_WEATHER_STORM },      // Thunderstorm
+    { -10, WeatherEffectType::Snow, 1, WeatherLevel::Light, SPR_WEATHER_SNOW },      // Snow
+    { -15, WeatherEffectType::Snow, 2, WeatherLevel::Heavy, SPR_WEATHER_SNOW },      // Heavy Snow
+    { -20, WeatherEffectType::Blizzard, 2, WeatherLevel::Heavy, SPR_WEATHER_SNOW },  // Blizzard
 };
 
 static constexpr const WeatherTransition ClimateTransitionsCoolAndWet[] = {
     { 8, 18, { 0, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 3, 3, 3, 4, 4, 0, 0, 0, 0, 0 } },
     { 10, 21, { 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3, 3, 3, 4, 4, 4, 5, 0, 0 } },
+    { 10, 23, { 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3, 3, 3, 4, 4, 4, 5, 6, 7 } },
```
Comment (Gymnasiast) [inline src/openrct2/world/Climate.cpp:456]: revert

**Comment by Gymnasiast** [comment]
> Glad to see it's been added sadly I don't have enough time to work on my own projects and open-source ones as well. I did try to reimplement it as much as possible from my first initial implementation way back. I'm also not involved in the project as much as I'd like to be to which come to a lot of discussion about certain types of weather. Hope someone makes good use of it and good luck to those that carry this on. +1

Thanks, and I completely understand. By all means, make sure your own projects (which pay the bills) are OK first, and don't feel too bad about yourself if there is no time or energy left for voluntary projects like this. They're voluntary, after all.

**Pull request merged**